config.splitting.js

```diff
@@ -67,22 +75,26 @@ module.exports = {
 
         const createRecord = ({ text = '', part }) => ({
           objectID: `${url.pathname} ${part}`,
-          path: url.pathname.split('/'),
           ...pageMeta,
           text,
         });
 
+        const serializeString = str => JSON.stringify(str).slice(1, -1); // remove surrounding quotes
+
+        const deserializeString = str => JSON.parse(`"${str}"`); // re-add surrounding quotes
+
         const splitToFitRecord = (text, baseRecord) => {
           const availableLength =
             MAX_RECORD_LENGTH - JSON.stringify(baseRecord).length;
+          const serializedText = serializeString(text); // needed to estimate the weight of that string, as it's gonna be stored in the Algolia record
```

Comment:
This the fix I was mentioning [there](https://github.com/algolia/crawler-configurations-examples/pull/3#discussion_r356048168):

> I also made the splitting logic work on JSON-serialized strings instead of pure strings, because the max record size was exceeded when special characters that need escaping (e.g. \") were found in the content.

I.e. if you had a paragraph that contains 10000 double-quote characters, it was counted as a record of size 10000, but in reality it was taking twice that size, because records are JSON-encoded and double quotes are encoded using two characters (`\"`) => when using the config on `body` instead of `p, li`, I was getting some `record_too_big` errors.

That's why I unfortunately had to add JSON serialization in the splitting logic itself, so that these costly escapable characters are correctly counted.

Reply:
Alright, mentioning the \" example in the comment can help to understand

Reply:
you're right! => done.